archive_feedstock.py

```diff
@@ -28,57 +38,108 @@ def archive_repo(owner, repo, archive=True, check_only=False):
     )
     raise_json_for_status(r)
 
-    target_status = f"{'' if archive else 'un'}archived"
+    if task == "archive":
+        target_status = "archived"
+    else:
+        target_status = "unarchived"
+
     data = r.json()
-    if data["archived"] == archive:
-        raise RuntimeError(f"{owner}/{repo} is already {target_status}!")
-    elif check_only:
+    if task == "archive" and data["archived"]:
+        print("feedstock %s is already %s" % (repo, target_status), flush=True)
+        return
+
+    if task == "unarchive" and not data["archived"]:
+        print("feedstock %s is already %s" % (repo, target_status), flush=True)
         return
 
     r = requests.patch(
         f"https://api.github.com/repos/{owner}/{repo}",
         headers=headers,
-        json={"archived": archive}
+        json={"archived": True if task == "archive" else False}
```

Comment:
```suggestion
        json={"archived": task == "archive"}
```

I prefer this shorter form, but if you think the longer one is more readable, that's ok.